sql/core/src/test/scala/org/apache/spark/sql/ui/UISeleniumSuite.scala

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.ui
+
+import scala.concurrent.duration._
+
+import org.apache.spark.sql.SQLContext
+import org.json4s.DefaultFormats
+import org.openqa.selenium.WebDriver
+import org.openqa.selenium.htmlunit.HtmlUnitDriver
+import org.scalatest.{BeforeAndAfterAll, Matchers}
+import org.scalatest.concurrent.Eventually._
+import org.scalatest.selenium.WebBrowser
+
+import org.apache.spark.{SparkContext, SparkConf, SparkFunSuite}
+import org.apache.spark.ui.SparkUICssErrorHandler
+
+/**
+ * Selenium tests for the SQL tab.
+ */
+class UISeleniumSuite extends SparkFunSuite with WebBrowser with Matchers with BeforeAndAfterAll {
+
+  implicit var webDriver: WebDriver = _
+  implicit val formats = DefaultFormats
+
+
+  override def beforeAll(): Unit = {
+    webDriver = new HtmlUnitDriver {
+      getWebClient.setCssErrorHandler(new SparkUICssErrorHandler)
+    }
+  }
+
+  override def afterAll(): Unit = {
+    if (webDriver != null) {
+      webDriver.quit()
+    }
+  }
+
+
+  /**
+   * Create a test SQLContext with the SparkUI enabled.
+   */
+  private def newSparkStreamingContext(): SQLContext = {
+    val conf = new SparkConf()
+      .setMaster("local")
+      .setAppName("test")
+      .set("spark.ui.enabled", "true")
+    val sc = new SparkContext(conf)
```

Comment:
@andrewor14, isn't this going to run into the same issue with multiple active SparkContexts that your patch hit? If that's the case, maybe we should bump up the priority on fixing that SQLContext testing issue?